### ISSUE TITLE
[BUG] Modify FLOPs in MFU calculation for casual mask when using FlashAttention.

### ISSUE THREAD

**Issue by Yuxin-CV** [NVIDIA/Megatron-LM#831]
Hi, I suggest we modify the [FLOPs calculation in the MFU](https://github.com/NVIDIA/Megatron-LM/blob/c3677e09aa4e2eec37048307bd795928b8f8324a/megatron/training/training.py#L88-L95) according to the [FlashAttention benchmark script](https://github.com/Dao-AILab/flash-attention/blob/9c0e9ee86d0e0022b60deddb405c20ab77481582/benchmarks/benchmark_flash_attention.py#L27-L30).

Specifically, the current calculation for the casual mask **can exceed 100% MFU** for seq_len = 16k (189 * 2 / 312 = 1.21), which is inaccurate. The FLOPs for the casual mask setting should be divided by 2 when using FlashAttention.

<img width="939" alt="flash2_a100_fwd_bwd_benchmark" src="https://github.com/pytorch/torchtitan/assets/57927171/56465aed-eafd-4618-943d-0ee39baac294">

**Comment by sbhavani** [comment]
Thanks for reporting this! This has been addressed in the codebase. The FLOPs calculation now properly accounts for causal attention by dividing by 2.
